app/src/main/java/tech/pauly/findapet/shared/MainActivity.java

```diff
@@ -61,6 +66,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
         switch (id) {
             case R.id.menu_search:
+                dataStore.save(new FilterAnimalTypeUseCase(viewModel.currentAnimalType.get()));
```

Comment:
This sucks. I'll admit it. I really want to have this function be delegated to the ViewModel somehow. I've tried solutions of how to do this in the past, basically piggybacking off of the Lifecycle architecture patterns, but I don't like it for this project. For now I'm just doing this here.